app/maptools/recordingmaptool.cpp

```diff
@@ -1228,6 +1228,7 @@ void RecordingMapTool::completeEditOperation()
 {
   if ( mActiveLayer && mActiveLayer->isEditCommandActive() )
   {
+    avoidIntersections();
```

Comment:
I believe this is only for (multi-)polygon layers, right? We can skip it for others